### PR TITLE
Support of ETS guard conditions for Registry.match

### PR DESCRIPTION
**Pull request by YurkoHoshko**
PR according to [Elixir Core Mailing List Proposal](https://groups.google.com/forum/#!topic/elixir-lang-core/SXNog8I-UYw)

### CODE REVIEW

File: lib/elixir/lib/registry.ex [hunk 1]
```diff
@@ -444,6 +444,11 @@ defmodule Registry do
   value or tuple element, while :"$1" can be used to temporarily assign part
   of pattern to a variable for a subsequent comparison.
 
+  It is possible to pass list of guard conditions for more precise matching.
+  Each guard is a tuple, which describes check that should be passed by assigned part of pattern.
+  For example :"$1" > 1 guard condition would be expressed as {:>, :"$1", 1} tuple.
+  Please note that guard conditions will work only for assigned variables like :"$1", :"$2", etc.
```
Comment (fishcakez) [inline lib/elixir/lib/registry.ex]: Please be explicit that `:"$_"` won't work as expected.

We could make it work, but lets not add the complexity until its required.

Comment (YurkoHoshko) [inline lib/elixir/lib/registry.ex]: Done.
I played around a little bit with it - it seems like in some cases you can use :"$_ in guard clauses without having assigned variables.

Example:
```
iex(1)> Registry.start_link(:duplicate, Registry.MatchTest)
{:ok, #PID<0.85.0>}
iex(2)> {:ok, _} = Registry.register(Registry.MatchTest, "hello", {1, :atom, 1})
{:ok, #PID<0.86.0>}
iex(3)> Registry.match(Registry.MatchTest, "hello", {:_, :_, :"$1"}, [{:is_tuple, :"$_"}]) |> Enum.sort()
[{#PID<0.81.0>, {1, :atom, 1}}]
iex(4)> Registry.match(Registry.MatchTest, "hello", {:_, :_, :"$1"}, [{:is_tuple, :"$$"}]) |> Enum.sort()
[]
iex(5)> Registry.match(Registry.MatchTest, "hello", {:_, :_, :_}, [{:is_tuple, :"$_"}]) |> Enum.sort()   
[{#PID<0.81.0>, {1, :atom, 1}}]
iex(6)> Registry.match(Registry.MatchTest, "hello", {:_, :_, :_}, [{:is_tuple, :"$$"}]) |> Enum.sort()
[]
```

Comment (fishcakez) [inline lib/elixir/lib/registry.ex]: Yeah, the issue is that `:"$_"` is for the whole term (`{key, {pid, value}}`) and not just `value`.

**Comment by fishcakez** [comment]
Thank you :heart:

File: lib/elixir/lib/registry.ex (hunk 1 above)
Comment (YurkoHoshko) [inline lib/elixir/lib/registry.ex]: yeah, that's true, I get your concern now. Thanks.

**Pull request merged**